src/org/ods/orchestration/util/ConcurrentCache.groovy

```diff
@@ -0,0 +1,33 @@
+package org.ods.orchestration.util
+
+import com.cloudbees.groovy.cps.NonCPS
+
+import java.util.concurrent.ConcurrentMap
+import java.util.function.Function
+
+class ConcurrentCache<K, V> {
+
+    private final ConcurrentMap<K, V> cache
+
+    ConcurrentCache(def cache) {
```

Comment:
Let's use a ConcurrentHashMap and don't let the user provide it to avoid possible concurrency issues. We could pass the constructor parameters for the map: initial size, load factor and/or concurrency level for writes.

Reply:
Done